Moved method from `PaletteManager` to extensions

Lehonti opened the pull request.

cameronwhite commented:
While I was testing this I noticed that the dialog can look pretty ugly - could you switch this to use the error dialog rather than the message dialog to match the regular "Unsupported file format" dialog when opening an invalid image file?

<img width="512" alt="Screenshot 2025-05-05 at 7 44 33 PM" src="https://github.com/user-attachments/assets/1c8eb7e7-cf75-40a9-9233-82e891cd8efc" />

vs

<img width="512" alt="Screenshot 2025-05-05 at 7 44 56 PM" src="https://github.com/user-attachments/assets/1dfde872-b0c8-4bd9-b862-594daf3b062b" />

Lehonti commented:
Sure, no problem @cameronwhite, I just switched to the error dialog.

At first I was disconcerted because I thought this was something I introduced during refactoring, but now I see what you meant.

cameronwhite commented:
Thanks!
